src/lua/init.lua

```diff
@@ -31,6 +31,18 @@ local LUA_TEMPLATES = { '?.lua', '?/init.lua' }
 local ROCKS_LIB_TEMPLATES = { ROCKS_LIB_PATH .. '/?.'..soext }
 local ROCKS_LUA_TEMPLATES = { ROCKS_LUA_PATH .. '/?.lua', ROCKS_LUA_PATH .. '/?/init.lua' }
 
+-- A proxy for pairs function that allows to use pairs for nested table in
+-- box.cfg.
+local builtin_pairs = rawget(_G, 'pairs')
+local function patched_pairs(object)
+    local mt = getmetatable(object)
+    local iterable = (mt and mt.__name == 'box_cfg') and
+                     debug.getmetatable(object).__index or
```

Comment:
Well, this part is not quite right.

1. It's unsafe, since `getmetatable` yields the value of `__metatable` field if one is set (that can be e.g. string[^string]). You need either to check the type of `mt` variable or to use `debug.getmetatable` to obtain `object`'s metatable. I'd rather use the latter approach.
[^string]: Fortunately, all Lua strings share the metatable with overloaded index, so `mt.__name` in case of strings just yields `nil`, but anyway in general case this is not safe.
2. There is no need in fetching `object`'s metatable one more time in the line above (it's already stored in `mt`, isn't it?) so just use `mt.__index` for this.